spacy/pipeline/_parser_internals/transition_system.pyx

```diff
@@ -155,6 +155,15 @@ cdef class TransitionSystem:
         action.do(state.c, action.label)
         state.c.history.push_back(action.clas)
 
+    def apply_actions(self, states, const int[::1] actions):
+        assert len(states) == actions.shape[0]
+        cdef StateClass state
+        cdef vector[StateC*] c_states
+        for state in states:
+            c_states.push_back(state.c)
```

Comment:
We can allocate the memory for the entire vector using `resize` and index-init the pointers to avoid the EH overhead of `push_back`.

Reply:
Great idea. Fixed.